### ISSUE TITLE
Crashes with AssertionError in Gson

### ISSUE THREAD

**Issue by freezy** [smooch/smooch-android#70]
Got four crashes so far in production:

```
java.lang.Enum$1.create (Enum.java:45)
com.google.gson.Gson.fromJson (Gson.java:700)
io.smooch.core.b.a.c.a (Unknown Source)
io.smooch.core.b.c.c (Unknown Source)
io.smooch.core.service.SmoochService.a (Unknown Source)
io.smooch.core.b.a (Unknown Source)
io.smooch.core.b$1.onServiceConnected (Unknown Source)
```

We're using v3.2.2 of the lib. We don't have ProGuard running which seemed to be the problem with #13. You can find more details [here](http://crashes.to/s/4fe15d303f5).

**Comment by wmora** [comment]
🤔 Like you said, this is usually caused by a missing Proguard config. Since you don't have it enabled, could you perhaps create a sample project with the same setup as your app reproducing the issue?

**Comment by freezy** [comment]
Unfortunately, I can't even reproduce it here locally, let alone isolate the problem in a sample project. The problem is that since it's executed on its own thread there is no way to wrap a try/catch around it to avoid crashing our app.

**Comment by wmora** [comment]
I believe this was solved in version [5.0.2](https://github.com/smooch/smooch-android/releases/tag/5.0.2). Feel free to reopen if it happens again
